client/src/components/ACL/GroupDetailsPane.js

```diff
@@ -39,41 +41,133 @@ export default class GroupDetailsPane extends React.Component {
+        const selected = !!existing && !!(existing.perm & mask);
+
+        return { existing, selected };
+    };
+
+    /*
+     * Toggles ACL for this predicate
+     * @param predicate - predicate to toggle ACL for
+     */
+    toggleACL = (predicate, mask) => {
+        const { group } = this.props;
+        const { existing, selected } = this.getPredicateACLStatus(
+            predicate,
+            mask,
+        );
+
+        if (!existing) {
+            group.acl.push({
+                predicate: predicate,
+                perm: mask,
+            });
+        } else {
+            existing.perm += !selected ? mask : -mask;
+        }
+    };
+
+    /*
+     * Saves ACL for this group
+     */
+    saveACL = () => {
+        const { group, saveNewAcl } = this.props;
+        saveNewAcl && saveNewAcl(group, group.acl);
+    };
+
+    /*
+     * Returns grid data for ACL predicates
+     * @return - Array of objects used to create grid
+     */
+    getGridData = () => {
+        const predicates = this.getFilteredACLPredicates();
+
         const getToggler = (p, mask) => {
-            const existing = group.acl.find(x => x.predicate === p);
-            const selected = !!existing && !!(existing.perm & mask);
+            const { selected } = this.getPredicateACLStatus(p, mask);
             return {
                 name: p,
                 selected,
                 toggle: () => {
-                    if (!existing) {
-                        group.acl.push({
-                            predicate: p,
-                            perm: mask,
-                        });
-                    } else {
-                        existing.perm += !selected ? mask : -mask;
-                    }
-                    saveNewAcl && saveNewAcl(group, group.acl);
+                    this.toggleACL(p, mask);
+                    this.saveACL();
                 },
             };
         };
 
-        return predicates
-            .filter(p => isAclPredicate(p.predicate))
-            .map(p => ({
-                name: p.predicate,
-                read: getToggler(p.predicate, ACL_READ),
-                modify: getToggler(p.predicate, ACL_MODIFY),
-                write: getToggler(p.predicate, ACL_WRITE),
-            }));
+        return predicates.map(p => ({
+            name: p.predicate,
+            read: getToggler(p.predicate, ACL_READ),
+            modify: getToggler(p.predicate, ACL_MODIFY),
+            write: getToggler(p.predicate, ACL_WRITE),
+        }));
     };
 
-    render() {
-        const { group, predicates, saveNewAcl } = this.props;
+    /*
+     * Renders the header with a 'Select All' checkbox
+     * @return - Function to render the header
+     */
+    getHeaderRenderer = mask => {
+        return ({ column }) => {
+            const predicates = this.getFilteredACLPredicates().map(
+                p => p.predicate,
+            );
+            const selectMode = predicates.every(
```

Comment:
"selectMode" isn't a descriptive name, because I expected it to mean "new select value" but in fact this const is true when all checkboxes are selected. It makes sense for its name to reflect that e.g. "const allSelected" or something like that